platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/paos/PaosInInterceptor.java

```diff
@@ -0,0 +1,445 @@
+                    .examineAllHeaderElements();
+            IDPEntry idpEntry = null;
+            String relayState = "";
+            String responseConsumerURL = "";
+            String messageId = "";
+            while (iterator.hasNext()) {
+                Element soapHeaderElement = (SOAPHeaderElement) iterator.next();
+                if (RELAY_STATE.equals(soapHeaderElement.getLocalName())) {
+                    relayState = DOM2Writer.nodeToString(soapHeaderElement);
+                } else if (REQUEST.equals(soapHeaderElement.getLocalName())
+                        && soapHeaderElement.getNamespaceURI()
+                        .equals(URN_OASIS_NAMES_TC_SAML_2_0_PROFILES_SSO_ECP)) {
+                    try {
+                        soapHeaderElement = SamlProtocol.convertDomImplementation(
+                                soapHeaderElement);
+                        Request ecpRequest = (Request) OpenSAMLUtil.fromDom(soapHeaderElement);
+                        IDPList idpList = ecpRequest.getIDPList();
+                        if (idpList == null) {
+                            throw new Fault(new AccessDeniedException(
+                                    "Unable to complete SAML ECP connection. Unable to determine IdP server."));
+                        }
+                        List<IDPEntry> idpEntrys = idpList.getIDPEntrys();
+                        if (idpEntrys == null || idpEntrys.size() == 0) {
+                            throw new Fault(new AccessDeniedException(
+                                    "Unable to complete SAML ECP connection. Unable to determine IdP server."));
+                        }
+                        //choose the right entry, probably need to do something better than select the first one
+                        //but the spec doesn't specify how this is supposed to be done
+                        idpEntry = idpEntrys.get(0);
+                    } catch (WSSecurityException e) {
+                        //TODO figure out IdP alternatively
+                        LOGGER.info(
+                                "Unable to determine IdP appropriately. ECP connection will fail. SP may be incorrectly configured. Contact the administrator for the remote system.");
+                    }
+                } else if (REQUEST.equals(soapHeaderElement.getLocalName())
+                        && soapHeaderElement.getNamespaceURI()
+                        .equals(URN_LIBERTY_PAOS_2003_08)) {
+                    responseConsumerURL = soapHeaderElement.getAttribute(RESPONSE_CONSUMER_URL);
+                    messageId = soapHeaderElement.getAttribute(MESSAGE_ID);
+                }
+            }
+            String token = createToken(authorization);
+            checkAuthnRequest(soapMessage);
+            Element authnRequestElement = SamlProtocol.getDomElement(soapMessage.getEnvelope()
+                    .getBody()
+                    .getFirstChild());
+            if (idpEntry == null) {
```

Comment:
Is there a reason we do not do this check before we create the token on line 190?  Neither of those methods can change its value and it looks like they should be skipped if an IDP entry cannot be found.